scripts/prepare-components.js

```diff
@@ -0,0 +1,206 @@
+
+/* eslint-env node */
+/* eslint-disable global-require */
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable import/no-unresolved */
+
+const documentationSiteLiveExamples = {
+`;
+
+// Returns a regular expression that can be used to search for specific exports in files
+const getExportRegExp = (line) => {
+	let formattedLine = line.replace(/\//g, '\\/');
+	formattedLine = formattedLine.replace(/\./g, '\\.');
+	return new RegExp(`${formattedLine}\r?\n?`);
+};
+
+// Make sure components are listed in index.js
+const checkIndex = indexFile
+	? (config) => {
+			let injectionIndex;
+			let injectionString = '';
+			let line;
+			let match;
+
+			// Component exports
+			if (config.component && config.classKey) {
+				// Check for SLDS-prefixed component export
+				line = `export SLDS${config.classKey} from './${config.component}';`;
+				match = indexFile.match(getExportRegExp(line));
+
+				if (match) {
+					injectionIndex = match.index + match[0].length;
+				} else {
+					injectionString += `${line}\n`;
+				}
+
+				// Check for non-prefixed component export
+				line = `export ${config.classKey} from './${config.component}';`;
+				match = indexFile.match(getExportRegExp(line));
+
+				if (match) {
+					injectionIndex = match.index + match[0].length;
+				} else {
+					injectionString += `${line}\n`;
+				}
+			}
+
+			// Sub-component exports
+			if (config.dependencies && config.dependencies.length > 0) {
+				config.dependencies.forEach((dependency) => {
+					if (config.component && dependency.component && dependency.classKey) {
+						// Check for SLDS-prefixed sub-component export
+						line = `export SLDS${dependency.classKey} from './${
+							config.component
+						}/${dependency.component}';`;
+						match = indexFile.match(getExportRegExp(line));
+
+						if (!match) {
+							injectionString += `${line}\n`;
+						}
+
+						// Check for non-prefixed sub-component export
+						line = `export ${dependency.classKey} from './${config.component}/${
+							dependency.component
+						}';`;
+						match = indexFile.match(getExportRegExp(line));
+
+						if (!match) {
+							injectionString += `${line}\n`;
+						}
+					}
+				});
+			}
+
+			// Add any missing parts
+			if (injectionString) {
+				if (!injectionIndex) {
+					injectionIndex = indexFile.search(/$(?![\r\n])/);
+				}
+				injectionIndex =
+					injectionIndex !== undefined ? injectionIndex : indexFile.length - 1;
+				indexFile = `${indexFile.slice(
+					0,
+					injectionIndex
+				)}${injectionString}\n${indexFile.slice(injectionIndex)}`;
+			}
+		}
+	: () => {};
+
+// Make sure components are listed in storybook-stories.js
+const checkStorybookStories = storybookStoriesFile
```

Comment:
Same suggestion here: if you used the AST from Babel your code could be shorter and more reliable since you're working with stronger guarantees instead of just string matches.

Reply:
Also, this one adds new ones at the end of the file, making them non-alphabetical once new ones are added.  If we used the Babel AST directly, we could add a step that could sort the exports when it writes the file back out.

Reply:
This is no longer needed as storybook-stories.js was removed with the Storybook 5 upgrade